numba_dpex/core/codegen.py

```diff
@@ -31,8 +31,8 @@ def _optimize_final_module(self):
         # Run some lightweight optimization to simplify the module.
         pmb = ll.PassManagerBuilder()
 
-        # Make optimization level depending on config.OPT variable
-        pmb.opt_level = config.OPT
+        # Make optimization level depending on config.DPEX_OPT variable
+        pmb.opt_level = config.DPEX_OPT
```

Comment:
We have to check here if the DPEX_OPT was set to `3` and then reset it to `2` and give the user a warning

Reply:
If it is by default 2 we can just warn user that it may be dangerous and lead to crashes. I'm thinking of example when user is hacking dpex and really want to see the output...